akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -289,7 +289,10 @@ private[http] object HttpServerBluePrint {
         def onPush(): Unit = {
           openTimeouts.head.clear()
           openTimeouts = openTimeouts.tail
-          push(responseOut, grab(responseIn))
+
+          // it may have been completed by a timeout already
```

Comment:
hm, I see it is just isAvailable and not isClosed. But then responseOut is not closed yet. So maybe I am just confused. Anyway, the fix is not self-evident ("it's your fault!" ;))